cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/TimePartitionedFileSetDatasetAvroSink.java

```diff
@@ -40,43 +43,64 @@
 import org.apache.hadoop.mapreduce.Job;
 
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  * A {@link BatchSink} to write Avro record to {@link TimePartitionedFileSet}
  */
+@Plugin(type = "sink")
+@Name("TimePartitionedFileSetDatasetAvroSink")
+@Description("AVRO Sink with Time Partitioned File Dataset")
 public class TimePartitionedFileSetDatasetAvroSink extends
   BatchSink<GenericRecord, AvroKey<GenericRecord>, NullWritable> {
 
-  private static final String TPFS_NAME = "name";
-  private static final String SCHEMA = "schema";
-  private static final String BASE_PATH = "basePath";
+  private static final String SCHEMA_DESC = "The schema of the record";
+  private static final String TPFS_NAME_DESC = "Name of the Time Partitioned FileSet Dataset to which the records " +
+    "have to be written";
+  private static final String BASE_PATH_DESC = "Optional: The base path for the time partitioned fileset. Should " +
```

Comment:
You don't have to say `Optional` here, since the meta info of the property in the REST endpoint already tells it's optional.